src/java.base/unix/classes/sun/nio/fs/UnixFileSystem.java

```diff
@@ -1037,6 +1045,15 @@ void copy(final UnixPath source,
             if (!flags.replaceExisting)
                 throw new FileAlreadyExistsException(
                     target.getPathForExceptionMessage());
+            else if (!sourceAttrs.isSymbolicLink() || flags.followLinks) {
+                // ensure source can be moved
+                try {
```

Comment:
This is copy rather than move but I think R_OK is correct, at least for regular files.